data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/GrokNamedCapturesUtilTest.java

```diff
@@ -0,0 +1,125 @@
+package org.opensearch.dataprepper.logstash.mapping;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.opensearch.dataprepper.logstash.mapping.GrokNamedCapturesUtil.GrokNamedCapturesPair;
+
+public class GrokNamedCapturesUtilTest {
+    private final String randomPrefix = UUID.randomUUID().toString();
+    private final String randomSuffix = UUID.randomUUID().toString();
+    private final String randomMiddle = UUID.randomUUID().toString();
+
+    private final List<String> namedCapturesPatterns = new ArrayList<>();
+    private final String firstNamedCapturesPattern =  UUID.randomUUID().toString();
+    private final String secondNamedCapturesPattern =  UUID.randomUUID().toString();
+
+    private final List<String> namedCapturesNames = new ArrayList<>();
+    private final String firstNamedCapturesName = UUID.randomUUID().toString();
+    private final String secondNamedCapturesName = UUID.randomUUID().toString();
+
+
+    @BeforeEach
+    public void setup() {
+        namedCapturesNames.add(firstNamedCapturesName);
+        namedCapturesNames.add(secondNamedCapturesName);
+
+        namedCapturesPatterns.add(firstNamedCapturesPattern);
+        namedCapturesPatterns.add(secondNamedCapturesPattern);
+    }
+
+    @Test
+    public void testSingleNamedCaptures() {
+        final String namedCapturesPattern =  UUID.randomUUID().toString();
+        final String namedCapturesName = UUID.randomUUID().toString();
+        final String regex = String.format("(?<%s>%s)", namedCapturesName, namedCapturesPattern);
+        final GrokNamedCapturesPair result = GrokNamedCapturesUtil.convertRegexNamedCapturesToGrokPatternDefinitions(regex);
+
+
+        assertThat(result.getMappedPatternDefinitions().size(), equalTo(1));
+
+        for (final Map.Entry<String, String> patternDefinition : result.getMappedPatternDefinitions().entrySet()) {
+           assertThat(patternDefinition.getValue().equals(namedCapturesPattern), equalTo(true));
+           final String expectedResult = String.format("%%{%s:%s}", patternDefinition.getKey(), namedCapturesName);
```

Comment:
`%%`? I was expecting `%{%s:%s}` but could be wrong.

Reply:
`%%` is how you do a literal `%` in String.format